tests/periph_i2c/main.c

```diff
@@ -20,270 +20,550 @@
+    if (argc == 5) {
+        flags = get_num(argv[4]);
+    }
+
+    printf("Command: i2c_write_reg(i2c_%i, 0x%02x, 0x%02x, 0x%02x, [0x%02x", i2c_dev,
+        addr, reg, flags, data);
+    puts("])");
+    res = i2c_write_reg(i2c_dev, addr, reg, data, flags);
+
+    if (res == I2C_ACK) {
+        printf("Success: i2c_%i wrote 1 byte\n", i2c_dev);
         return 0;
     }
+    return print_i2c_error(res);
 }
 
-int cmd_read_reg(int argc, char **argv)
+int cmd_i2c_write_regs(int argc, char **argv)
 {
     int res;
-    uint8_t addr, reg;
-    int length;
+    uint16_t addr;
+    uint16_t reg;
+    uint8_t flags = 0;
+    int len = argc - 4;
     uint8_t data[BUFSIZE];
 
-    if (i2c_dev < 0) {
-        puts("Error: no I2C device initialized");
+    if (check_dev() == 1){
         return 1;
     }
-    if (argc < 4) {
-        puts("Error: not enough arguments given");
-        printf("Usage:\n%s ADDR REG LENGTH\n", argv[0]);
+
+    if (argc < 5) {
+        INVALID_ARGS;
+        printf("Usage:\n%s ADDR REG FLAG BYTE0 [BYTE1 ...]\n", argv[0]);
         return 1;
     }
 
-    addr = atoi(argv[1]);
-    reg = atoi(argv[2]);
-    length = atoi(argv[3]);
-
-    if (length < 1 || length > (int)BUFSIZE) {
-        puts("Error: invalid LENGTH parameter given");
-        return 1;
+    addr = get_num(argv[1]);
+    reg = get_num(argv[2]);
+    flags = get_num(argv[3]);
+    for (int i = 0; i < len; i++) {
+        data[i] = get_num(argv[i + 4]);
     }
-    else if (length == 1) {
-        printf("i2c_read_reg(I2C_%i, 0x%02x, 0x%02x)\n", i2c_dev, addr, reg);
-        res = i2c_read_reg(i2c_dev, addr, reg, data);
+
+    printf("Command: i2c_write_regs(i2c_%i, 0x%02x, 0x%02x, 0x%02x, [", i2c_dev,
+        addr, reg, flags);
+    for (int i = 0; i < len; i++) {
+        if (i != 0) {
+            printf(", ");
+        }
+        printf("0x%02x", data[i]);
     }
-    else {
-        printf("i2c_read_regs(I2C_%i, 0x%02x, 0x%02x, %i)\n", i2c_dev, addr, reg, length);
-        res = i2c_read_regs(i2c_dev, addr, reg, data, length);
+    puts("])");
+    res = i2c_write_regs(i2c_dev, addr, reg, data, len, flags);
+
+    if (res == I2C_ACK) {
+        printf("Success: i2c_%i wrote %i bytes to reg 0x%02x\n",
+            i2c_dev, len, reg);
+        return 0;
     }
+    return print_i2c_error(res);
+}
 
-    if (res < 1) {
-        puts("Error: no bytes were read");
+int cmd_i2c_get_devs(int argc, char **argv)
+{
+    (void)argv;
+    (void)argc;
+    printf("Success: Amount of i2c devices: [%d]\n", I2C_NUMOF);
+    return 0;
+}
+
+int cmd_i2c_get_speed(int argc, char **argv)
+{
+    (void)argv;
+    (void)argc;
+    if (check_dev() == 1){
         return 1;
     }
-    else {
-        printf("I2C_%i: successfully read %i bytes from reg 0x%02x:\n  [", i2c_dev, res, reg);
-        for (int i = 0; i < res; i++) {
-            printf("0x%02x, ", data[i]);
-        }
-        puts("]");
-        return 0;
+
+    switch (GET_I2C_CLK_SPD(i2c_dev)) {
```

Comment:
The indenting of this switch/case statement is somewhat unusual. The Coding Convention does not say anything specific about it, but I like to see an extra indentation for the statements besides the "case".

    switch (GET_I2C_CLK_SPD(i2c_dev)) {
        case case I2C_SPEED_LOW:
            printf("Success: i2...");
            break;
        case I2C_SPEED_NORMAL: